Extractor/ServerSubscriptionManager.cs

```diff
@@ -0,0 +1,153 @@
+
+        public ServerSubscriptionManager(
+            ILogger<ServerSubscriptionManager> logger,
+            UAClient uaClient,
+            RebrowseTriggersConfig config,
+            UAExtractor extractor
+        )
+        {
+            _logger = logger;
+            _uaClient = uaClient;
+            _config = config;
+            _extractor = extractor;
+        }
+
+        public async Task EnableCustomServerSubscriptions(CancellationToken token)
+        {
+            _logger.LogInformation("{startTime}", UAExtractor.StartTime);
+            var targetNodes = _config.Targets.GetValues;
+            List<NodeId> nodeIds = new List<NodeId>();
+            var filteredNamespaces = _config.Namespaces;
+            var filteredNamespacesCount = filteredNamespaces?.Count();
+            var shouldFilterNamespaces = filteredNamespacesCount > 0;
+
+            var serverNamespaces = ObjectIds.Server_Namespaces;
+
+            var grouping = new Dictionary<NodeId, List<ReferenceDescription>>();
+            // displayName: nodeId
+            var namespaceNameToId = new Dictionary<string, string>();
+
+            await _uaClient.Browser.BrowseDirectory(
+                new[] { serverNamespaces },
+                (refDef, parent) =>
+                {
+                    var nodeId = (NodeId)refDef.NodeId;
+
+                    if (parent == serverNamespaces && !grouping.ContainsKey(nodeId))
+                    {
+                        grouping.Add(nodeId, new List<ReferenceDescription>());
+                        namespaceNameToId.Add(refDef.DisplayName.ToString(), nodeId.ToString());
+                    }
+                    else if (
+                        grouping.ContainsKey(parent)
+                        // Ensures that the type of node being added is a variable node class
+                        && refDef.NodeClass == NodeClass.Variable
+                        // Filters targets nodes
+                        && targetNodes.Contains(refDef.DisplayName.ToString())
+                    )
+                    {
+                        grouping.GetValueOrDefault(parent).Add(refDef);
+                    }
+                },
+                token,
+                maxDepth: 1,
+                doFilter: false,
+                ignoreVisited: false
+            );
+
+            // To be used in filtering namespaces
+            var availableNamespaces = namespaceNameToId.Keys;
+
+            // Filters by namespaces
+            var processedNamespaces = (
+                shouldFilterNamespaces
+                    ? filteredNamespaces.Intersect(availableNamespaces)
+                    : availableNamespaces
+            ).ToList();
+
+            if (shouldFilterNamespaces && processedNamespaces.Count() < filteredNamespacesCount)
+            {
+                _logger.LogInformation(
+                    "Some namespaces could not be processed as they do not exist on the server: {namespaces}",
```

Comment:
Maybe use something other than "processesed"? Remember that users generally don't have the context of where the log message comes from.